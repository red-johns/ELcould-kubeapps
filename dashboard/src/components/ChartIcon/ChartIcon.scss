.ChartIcon {
  height: 64px;
}
.ChartIcon__img {
  max-height: 64px;
}
