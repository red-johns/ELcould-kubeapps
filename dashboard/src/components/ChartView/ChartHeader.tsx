import * as React from "react";
import { Link } from "react-router-dom";

import ChartIcon from "../ChartIcon";

interface IChartHeaderProps {
  appVersion?: string;
  id: string;
  icon?: string;
  repo: string;
  description: string;
}

class ChartHeader extends React.Component<IChartHeaderProps> {
  public render() {
    const { appVersion, id, icon, repo, description } = this.props;
    return (
      <header>
        <div className="ChartView__heading margin-normal">
          <ChartIcon icon={icon} />
          <div className="title margin-l-small">
            <h1 className="margin-t-reset">{id}</h1>
            <h5 className="subtitle margin-b-normal">
              {appVersion && <span>{appVersion} - </span>}
              <Link to={`/catalog/${repo}`}>{repo}</Link>
            </h5>
            <h5 className="subtitle margin-b-reset">{description}</h5>
          </div>
        </div>
        <hr />
      </header>
    );
  }
}

export default ChartHeader;
