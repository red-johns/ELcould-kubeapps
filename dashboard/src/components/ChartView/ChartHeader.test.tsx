import { shallow } from "enzyme";
import * as React from "react";
import { Link } from "react-router-dom";

import ChartIcon from "../ChartIcon";
import ChartHeader from "./ChartHeader";

const testProps: any = {
  description: "A Test Chart",
  id: "testrepo/test",
  repo: "testrepo",
};

it("renders a header for the chart", () => {
  const wrapper = shallow(<ChartHeader {...testProps} />);
  expect(wrapper.text()).toContain("testrepo/test");
  expect(wrapper.text()).toContain("A Test Chart");
  const repoLink = wrapper.find(Link);
  expect(repoLink.exists()).toBe(true);
  expect(repoLink.props()).toMatchObject({ to: "/catalog/testrepo", children: "testrepo" });
  expect(wrapper.find(ChartIcon).exists()).toBe(true);
  expect(wrapper).toMatchSnapshot();
});

it("displays the appVersion", () => {
  const wrapper = shallow(<ChartHeader {...testProps} appVersion="1.2.3" />);
  expect(wrapper.text()).toContain("1.2.3");
});

it("uses the icon", () => {
  const wrapper = shallow(<ChartHeader {...testProps} icon="test.jpg" />);
  const icon = wrapper.find(ChartIcon);
  expect(icon.exists()).toBe(true);
  expect(icon.props()).toMatchObject({ icon: "test.jpg" });
});
