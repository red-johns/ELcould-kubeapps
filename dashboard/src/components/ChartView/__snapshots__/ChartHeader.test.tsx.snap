// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a header for the chart 1`] = `
<header>
  <div
    className="ChartView__heading margin-normal"
  >
    <ChartIcon />
    <div
      className="title margin-l-small"
    >
      <h1
        className="margin-t-reset"
      >
        testrepo/test
      </h1>
      <h5
        className="subtitle margin-b-normal"
      >
        <Link
          replace={false}
          to="/catalog/testrepo"
        >
          testrepo
        </Link>
      </h5>
      <h5
        className="subtitle margin-b-reset"
      >
        A Test Chart
      </h5>
    </div>
  </div>
  <hr />
</header>
`;
